src/EnergyPlus/api/datatransfer.py

```diff
@@ -248,6 +279,25 @@ def __init__(self, api: cdll, running_as_python_plugin: bool = False):
         self.api.currentSimTime.argtypes = [c_void_p]
         self.api.currentSimTime.restype = RealEP
 
+    def get_api_data(self, state: c_void_p) -> List[APIDataExchangePoint]:
+        """
+        Returns a nicely formed list of API data exchange points available in the current simulation.
+
+        :param state: An active EnergyPlus "state" that is returned from a call to `api.state_manager.new_state()`.
+        :return: Returns a Python list of APIDataExchangePoint instances, which can be filtered to inspect exchanges.
+        """
+        count = c_int()
+        r = self.api.getAPIData(state, byref(count))
+        list_response = [
+            DataExchange.APIDataExchangePoint(
+                r[i].what.decode('utf-8'),
+                r[i].name.decode('utf-8'),
+                r[i].key.decode('utf-8'),
+                r[i].type.decode('utf-8')) for i in range(count.value)
+        ]
+        self.api.freeAPIData(r, count)  # free the underlying C memory now that we have a Python copy
+        return list_response
```

Comment:
On the Python API, we only expose one function.  This function handles the whole process of reading from the raw C API, copying into nice Python instances, and freeing the C API memory.  This results in a super slick and familiar experience for the users.